tic/tic.go

```diff
@@ -0,0 +1,1330 @@
+	v, err := d.getVar8(OffsetInputState)
+	return InputState(v), err
+}
+
+// GetInputAfterAveraging gets a variable used in the process that converts raw
+// RC and analog values into a motor position or speed. This is mainly for
+// debugging your input scaling settings in RC or analog mode.
+//
+// A value of tic.InputNull means the input value is not available.
+func (d *Dev) GetInputAfterAveraging() (uint16, error) {
+	return d.getVar16(OffsetInputAfterAveraging)
+}
+
+// GetInputAfterHysteresis gets a variable used in the process that converts raw
+// RC and analog values into a motor position or speed. This is mainly for
+// debugging your input scaling settings in RC or analog mode.
+//
+// A value of tic.InputNull means the input value is not available.
+func (d *Dev) GetInputAfterHysteresis() (uint16, error) {
+	return d.getVar16(OffsetInputAfterHysteresis)
+}
+
+// GetInputAfterScaling gets the value of the Tic's main input after scaling has
+// been applied.
+//
+// If the input is valid, this number is the target position or target velocity
+// specified by the input.
+func (d *Dev) GetInputAfterScaling() (int32, error) {
+	v, err := d.getVar32(OffsetInputAfterScaling)
+	return int32(v), err
+}
+
+// MotorDriverError describes the possible motor driver errors for the Tic T249.
+type MotorDriverError uint8
+
+const (
+	MotorDriverErrorNone            MotorDriverError = 0
+	MotorDriverErrorOverCurrent     MotorDriverError = 1
+	MotorDriverErrorOverTemperature MotorDriverError = 2
+)
+
+// GetLastMotorDriverError gets the cause of the last motor driver error.
+//
+// This is only valid for the Tic T249.
+func (d *Dev) GetLastMotorDriverError() (MotorDriverError, error) {
+	if d.variant != TicT249 {
+		return 0, ErrUnsupportedVariant
+	}
+
+	v, err := d.getVar8(OffsetLastMotorDriverError)
+	return MotorDriverError(v), err
+}
+
+// GetLastHPDriverErrors gets the "Last HP driver errors" variable.
+//
+// Each bit in this register represents an error. If the bit is 1, the error was
+// one of the causes of the Tic's last motor driver error.
+//
+// This is only valid for the Tic 36v4.
+func (d *Dev) GetLastHPDriverErrors() (uint8, error) {
+	if d.variant != Tic36v4 {
+		return 0, ErrUnsupportedVariant
+	}
+
+	return d.getVar8(OffsetLastHPDriverErrors)
+}
+
+// GetSetting gets a contiguous block of settings from the Tic's EEPROM.
+//
+// The maximum length that can be fetched is 15 bytes.
+//
+// This library does not attempt to interpret the settings and say what they
+// mean. If you are interested in how the settings are encoded in the Tic's
+// EEPROM, see the "Settings reference" section of the Tic user's guide.
+func (d *Dev) GetSetting(offset Offset, length uint) ([]uint8, error) {
+	if length > 15 {
+		return nil, errors.New("maximum length exceeded")
+	}
+
+	return d.getSegment(cmdGetSetting, offset, length)
+}
+
+// Offset represents where settings are stored in the Tic's EEPROM memory. See
```

Comment:
Unless mistaken, I didn't see the type "Offset" being used in any of the APIs. As such, I'd prefer to not export it as this will confuse the user. That's the only change I'd prefer to see before merging.

Reply:
You're right - fixed in b0dd501.